packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -265,18 +172,33 @@ export class FacetSearch {
     );
   }
 
-  private get suggestionList() {
+  private get resultList() {
+    return this.facetSearchResults.map((searchResult, index) => (
+      <li
+        onClick={() => this.onSelectValue(index)}
+        onMouseDown={(e) => e.preventDefault()}
+        part="search-result"
+        class={FacetSearch.searchResultClasses}
+        value={index}
+        aria-label={this.component.ariaLabelForSearchResult(searchResult)}
+      >
+        {this.component.renderSearchResult(searchResult)}
```

Comment:
I would have though we would completely separate the "search input" (the file facet-search.tsx, which could be renamed to facet-search-input) from the "suggestion/list of choices". 

And, in the parent Category or standard facet component, we could do: 
(all pseudo code...)
```
init() {
  this.facetSearchInput = new FacetSearchInput();

  // or new CategoryFacetSearchSuggestions();
  this.facetSearchSuggestions = new StandardFacetSearchSuggestions(); 
  
  this.facetSearchSuggestions.onSelect = (value)=> this.facetSearchInput.submit(value);
}

render() {
 return (
     [...]
     {this.facetSearchInput.render()}
     {this.facetSearchSuggestions.render()}
  )
}
```

That way it seems we get more of a pattern where the parent is orchestrating both sub classes, versus a weird-ish back and forth between the two ?


Do you think it makes sense ?

Reply:
It uses the combobox which is an accessibility helper class that manages attributes for an input + a list of suggestion

Reply:
renderSearchResult is here just to render the bit of content inside a suggestion (li element) which has to manage various accessibility attributes